howso/engine/trainee.py

```diff
@@ -188,6 +186,10 @@ def id(self) -> str | None:
         """
         return self._id
 
+    @id.setter
+    def id(self, id: str) -> None:
```

Comment:
I believe the setter was excluded here since you wouldn't ever change it after creation. (Previously the base class implemented both, but by overriding only the getter here it prevented setting it)

Reply:
Addressed by: c28e6de8dded39f04ba3206735fe885f931d8d2e